tests/unit/vault/test_shares.py

```diff
@@ -498,3 +498,90 @@ def test__mint_shares_with_zero_total_supply_positive_assets(
     # shares should be minted at 1:1
     assert vault.balanceOf(fish) == amount
     assert vault.pricePerShare() > (10 ** vault.decimals())
+
+
+def test__mint_shares_with_zero_total_supply_positive_assets(
+    asset, fish_amount, fish, initial_set_up, gov
+):
+    amount = fish_amount // 10
+    first_profit = fish_amount // 10
+
+    vault, strategy, _ = initial_set_up(asset, gov, amount, fish)
+    create_profit(asset, strategy, gov, vault, first_profit)
+    vault.update_debt(strategy, int(0), sender=gov)
+    assert (
+        vault.totalSupply() > amount
+    )  # there are more shares than deposits (due to profit unlock)
+
+    # User redeems shares
+    vault.redeem(vault.balanceOf(fish), fish, fish, sender=fish)
+
+    assert vault.totalSupply() > 0
+
+    ape.chain.mine(timestamp=ape.chain.pending_timestamp + 14 * 24 * 3600)
+
+    assert vault.totalSupply() == 0
+
+    vault.mint(amount, fish, sender=fish)
+
+    # shares should be minted at 1:1
+    assert vault.balanceOf(fish) == amount
+    assert vault.pricePerShare() > (10 ** vault.decimals())
+
+
+def test__deposit_with_zero_total_assets_positive_supply(
+    asset, fish_amount, fish, initial_set_up, gov
+):
+    amount = fish_amount // 10
+
+    vault, strategy, _ = initial_set_up(asset, gov, amount, fish)
+
+    # Create a loss
+    asset.transfer(gov, amount, sender=strategy)
+    strategy.report(sender=gov)
+
+    assert strategy.convertToAssets(amount) == 0
+
+    vault.process_report(strategy, sender=gov)
+
+    assert vault.totalAssets() == 0
+    assert vault.totalSupply() != 0
+
+    with ape.reverts("cannot mint zero"):
+        vault.deposit(amount, fish, sender=fish)
+
+    # shares should not be
```

Comment:
Shares should not be minted?

Reply:
https://github.com/yearn/yearn-vaults-v3/pull/209/commits/6a5ead1ffc7be7d4e4614574f6519b41996cd297